### PR TITLE
1273/stablecoin selection pricing

### LINKED ISSUE #1273: Smart selection of quote token
Smart selection of quote token and base token

Rafa's suggestion for simple and good enough solution is:
- We need to know when a token is a stable coin
- If it's stable coin vs non stable coin, we choose stable coin as a quote
- In any other case, we continue as now: sell is the quote

With the above, we can eliminate the need of showing the 2 prices.
We can, just in case allow to invert the price anyways.

_Estimation: 3.5_

### PR DESCRIPTION
**Pull request by W3stside**
Will close #1273 

Makes check inside `useTokenList` via `Map<TokenAddress, Priority>` lookup and assigns a `priority: number` on `TokenDetails`

WIP

Will now check inside Price what to show via prio

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1344--dexreact.review.gnosisdev.com

**Comment by W3stside** [comment]
Need to fix price when swapping tokens. Should be inverted price but still show quote token, where appropriate

File: src/data/stablecoinList.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+export default [
+  {
+    // USD coins
+    priority: 1,
+    addresses: {
+      1: [
```
Comment (anxolin) [inline src/data/stablecoinList.ts:6]: Do not hardcode the network. We have an enum, we can say MAINNET here

File: src/hooks/useTokenList.ts [hunk 2]
```diff
@@ -3,21 +3,52 @@ import { TokenDetails } from 'types'
...
 export interface UseTokenListParams {
   networkId?: number
   excludeDeprecated?: boolean
+  stablecoinList?: typeof stablecoins
 }
 
-export const useTokenList = ({ networkId, excludeDeprecated }: UseTokenListParams = {}): TokenDetails[] => {
+function getNetworkCorrectStableCoinList(
+  networkId: number | undefined,
+  stablecoinList: typeof stablecoins,
```
Comment (anxolin) [inline src/hooks/useTokenList.ts:16]: we said this is not a stable coin anymore, note we have for example WETH.
It's a `quoteTokenPriorities` or something similar

**Review by anxolin — CHANGES_REQUESTED** [review]
We should use by default the QUOTE to give the price, that' by definition. However this PR uses the base token.

image


Also, OWL is is missing in our list. We shouldn't forget our own stable coin :) 

Some concerns in the code, the biggest, the implementation is not very general, and we should add unit testing for the function's logic.

File: src/hooks/usePrioritiseTokensForPrice.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+import { TokenDetails } from 'types'
+
+interface HookProps {
+  sellToken: TokenDetails
+  receiveToken: TokenDetails
+  isInverse: boolean
+}
+
+export default ({ sellToken, receiveToken, isInverse }: HookProps): boolean => {
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/hooks/usePrioritiseTokensForPrice.ts:10]: did you not look at the testing? :facepalm:

File: src/data/stablecoinList.ts (hunk 1 above)
Comment (W3stside) [reply src/data/stablecoinList.ts:6]: this is coming directly from the JS file i was given, i think 1 is better to grab with our networkId

File: src/hooks/usePrioritiseTokensForPrice.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  // baseToken = receiveToken
+  // quoteToken = sellToken
+  // baseToken/quoteToken
+  const showInverse = useMemo(() => {
+    const receiveIsWeth = receiveToken.priority === 3
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/hooks/usePrioritiseTokensForPrice.ts:15]: :+1:

File: src/hooks/usePrioritiseTokensForPrice.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (W3stside) [reply src/hooks/usePrioritiseTokensForPrice.ts:10]: ah nvm different fn, will do

**Comment by W3stside** [comment]
closing in favour of #1350

**Pull request closed without merging**